### ISSUE TITLE
Using reserved keyword in record field

### ISSUE THREAD

**Issue by devmaars** [teal-language/tl#746]
Hi, i was asking myself, if my record i want to use a keyword that teal grammar already use, how can i escape ? in lua for example we can use string encapsulation
```lua
local t = {}
t.end = 200 --- Error
t["end"] = 200 --- work
```

So here come record data type

```lua
local record MemoryRegion
    state: integer
    start: integer
    end: integer --- error
    type: integer --- no error even tho type is a defined function in lua/teal
    name: string
    internalName: string
end
```
So my question is how can i use a reserved keyword in a record ? is there a way to escape string ?

**Comment by hishamhm** [comment]
You can do it just like Lua table declarations.

```lua
local record MemoryRegion
    state: integer
    start: integer
    ["end"]: integer
    type: integer
    name: string
    internalName: string
end
```

Like in Lua tables, only Lua keywords need this kind of "escaping". Keywords added by Teal (e.g. `record`, `global`, etc.) are "soft keywords", so they do not require escaping.

**Comment by devmaars** [comment]
Thanks, seem working but it make the vscode go crazy
![image](https://github.com/teal-language/tl/assets/163019521/4dc52a05-b0b1-4f28-8049-287f041da9b5)

**Comment by hishamhm** [comment]
Thanks for the feedback!

@devmaars does it report an error message?

@pdesaulniers any idea on the vscode-teal issue above?

**Comment by devmaars** [comment]
No there is no error, just the color that become all green.
I have also tried disabling all my extensions to ensure that it is vs code teal, causing the color issue

**Comment by pdesaulniers** [comment]
@devmaars The syntax highlighting issue should be fixed in vscode-teal v0.10.0. Bracket notation was not supported inside record declarations.

![2024-08-25-10:44:38](https://github.com/user-attachments/assets/337f823e-2a9c-4a7b-aaec-290dbfad7606)
